### ISSUE TITLE
Selection area is invisible after update to 1.4.0

### ISSUE THREAD

**Issue by dsechin** [Simonwep/selection#33]
#### Do you want to request a **feature** or report a **bug**? 

a **bug**

#### What is the current behavior?

After updating Selection to any version, greater then `0.1.4`, I do not see the selection area on selection.

#### What is the expected behavior?

It is quite difficult to extract the page I'm using Selection on out of the whole project.

But it seems to me that change is behaviour is caused by the `that._clippingElement` added after 0.1.4

**How it worked before (v0.1.0)**:
![2019-04-16 13 48 16](https://user-images.githubusercontent.com/26961701/56203679-60d81400-604e-11e9-92a4-3e6fddc92bdd.gif)

**How it works now (v0.1.5, same on 0.2.1)**:
![2019-04-16 13 47 35](https://user-images.githubusercontent.com/26961701/56203713-70eff380-604e-11e9-9aa4-316ad512fd7a.gif)

Are there any restrictions on the style of `html`/`body` tags?

#### Your environment:
```
Selection-version: 0.1.5
Browser-version:  Chrome Version 73.0.3683.103
Operating-system:  macOS Mojave 10.14.4
```

**Comment by Simonwep** [comment]
I'm not aware of any, there was #29 where `selectables` was removed. The `html` and `body`  elements shouldn't be used as boundaries since selectionjs should be restricted to a specific area.

No problem, but cuold you provide your selection.js config? I'll try to reproduce it.

`_clippingElement` is used to restrict the visible area if the container element is scrollable.
Minor version indicated some kind of braking changes, I'll try to fix these as soon as possible :)

**Comment by dsechin** [comment]
Wow, that was fast!

Thank you for the reply.

My config is as follows: 

```
const selectionOptions = {

            class: 'policy-consumers-selection-area',

            selectables: ['.policy-consumers-list__consumer'],
            singleClick: false, // handled by the ng-click callback
            disableTouch: true,
            boundaries: ['.policies-editor-content__right', '.policies-editor-footer__right'],

            onSelect(evt) {
                // Check if clicked element is already selected
                const selected = evt.target.classList.contains(selectedConsumerClass);

                if (!evt.originalEvent.shiftKey) {
                    // Remove class from every element which is selected
                    evt.selectedElements.forEach(s => s.classList.remove(selectedConsumerClass));

                    // Clear previous selection
                    this.clearSelection();
                }

                if (selected) {
                    // Unselect element
                    evt.target.classList.remove(selectedConsumerClass);
                    this.removeFromSelection(evt.target);
                } else {
                    // Select element
                    evt.target.classList.add(selectedConsumerClass);
                    this.keepSelection();
                }
            },

            onStart(evt) {
                if (vm.dragStarted) {
                    // selection.cancel() is a wrapper
                    // for the selection._onTapStop()
                    // which throws an error almost every time (due to a bug)
                    this._onTapStop(evt);

                    const areaEl = evt.areaElement;

                    if (areaEl) {
                        areaEl.hidden = true;

                        $timeout(() => {
                            areaEl.hidden = false;
                        });
                    }

                    return;
                }

                // Remove class if the user don't pressed the shift key
                if (!evt.originalEvent.shiftKey) {
                    vm.deselectSelectedConsumers();
                    this.clearSelection();
                }
            },

            onStop(evt) {
                [...evt.selectedElements].forEach(x => {
                    const $e = angular.element(x);

                    const consumer = _.get(
                        vm,
                        [
                            'policiesToShowConsumersFrom',
                            Number($e.attr('policy-idx')),
                            $e.attr('item-prop'),
                            Number($e.attr('item-idx'))
                        ]
                    );

                    if (!consumer) {
                        return;
                    }

                    const consumerId = consumer[constants.ID_PROP_BY_POLICY_PROP[$e.attr('item-prop')]];

                    consumer.isSelected = !consumer.isSelected;

                    if (consumer.isSelected) {
                        vm.selectedConsumers[consumerId] = consumer;
                        vm.selectedConsumersCount++;
                    } else {
                        delete vm.selectedConsumers[consumerId];
                        vm.selectedConsumersCount--;
                    }
                });

                this.clearSelection();

                if ($scope.$root.$$phase !== '$apply' && $scope.$root.$$phase !== '$digest') {
                    $scope.$apply();
                }
            }
        };
```

and `.policy-consumers-selection-area`:

```
.policy-consumers-selection-area {
    background: rgba(0, 0, 255, 0.1);
    border-radius: 0.1em;
    border: 0.05em solid rgba(0, 0, 255, 0.2);
    z-index: 20;
}
```

Your lib is really **awesome**, by the way.

**Comment by Simonwep** [comment]
You're welcome :)
Okay, that's strange... If mounted, could you send me a screenshot of the selection-area container which is responsible for the clipping? After this I can investigate why it's clipped / invisible :) Like so:

![Screenshot (104)](https://user-images.githubusercontent.com/30767528/56215848-a43b7e00-6060-11e9-911f-efe0ee827bc4.png)
**red** - _Clipping element_
**green** - _Actual selection area_

**Comment by dsechin** [comment]
Here's the clipping element on page load:

![image](https://user-images.githubusercontent.com/26961701/56216805-c76b2b00-606a-11e9-89f9-68c8621829e3.png)

And here it is after selecting a bunch of elements:
![image](https://user-images.githubusercontent.com/26961701/56216937-039e8b80-606b-11e9-89bf-8872d8002769.png)

Selection area after selecting a bunch of elements:
![image](https://user-images.githubusercontent.com/26961701/56217031-30eb3980-606b-11e9-8e62-788b7bda5108.png)

I can provide you with a remote access to the working demo if that's ok

**Comment by Simonwep** [comment]
Yeah, solving this problem directly on the application itself would be the fastest way to examine the problem. This would be great! Furthermore, if the second image was taken after a selection (the area element has `display: none`), there's at the first view nothing suspicious in your Environment which would cause this problem.

**Comment by dsechin** [comment]
I think I've got it working. Please try to open the following URL: [demo.selection.keenetic.limited](https://demo.selection.keenetic.limited).

If it opens successfully (that can take a bit of time & reloads) use `selection` as login and as password.
I use Selection on the second tab ("Policy bindings") of the [`controlPanel/policies` page](https://demo.selection.keenetic.limited/controlPanel/policies).

**Comment by dsechin** [comment]
This is a temporary build, living on my machine. I can make a more permanent solution if this one is too hard to use. Thank you once again for wanting to help.

**Comment by Simonwep** [comment]
Well, thank you for providing this demo and great communication to find it :) I found the problem, the clipping area don't have any `z-index` property, so the applied `z-index` on the selection-area element don't even take effect. After adding `z-index: 1` to the clipping element it works. 

Until I published a new version (will be very soon since this looks like a serious bug to me) you can use this piece of code in your selectionjs config to fix this: 

```js
onStart({selection}) {
        selection._clippingElement.style.zIndex = 1;
}
```

**Comment by dsechin** [comment]
> Well, thank you for providing this demo and great communication to find it :) I found the problem, the clipping area don't have any `z-index` property, so the applied `z-index` on the selection-area element don't even take effect. After adding `z-index: 1` to the clipping element it works.
> 
> Until I published a new version (will be very soon since this looks like a serious bug to me) you can use this piece of code in your selectionjs config to fix this:
> 
> ```js
> onStart({selection}) {
>         selection._clippingElement.style.zIndex = 1;
> }
> ```

WoW! This is so great ^_^

Thank you so much. I can't express how excited I am 🔥

**Comment by Simonwep** [comment]
Oh, GitHub discovered the fix-commit: dd9926bae52dad5f3c8eeb5a026b9c89dca4bdaa :D

**Comment by Simonwep** [comment]
@dsechin Thought I'll inform you I published [v0.2.2](https://www.npmjs.com/package/@simonwep/selection-js) where this fix takes action :)

**Comment by dsechin** [comment]
@Simonwep I know, that this issue is closed, but I would like to point out that I get the same behavior in my setup with the latest release.

I found a simple fix for it: wrap `zIndex` value in quotes:

```
            css(that._clippingElement, {
                overflow: 'hidden',
                position: 'fixed',
                transform: 'translate3d(0, 0, 0)', // https://stackoverflow.com/a/38268846
                pointerEvents: 'none',
                zIndex: '1' // <----------------------
            });
```

This way it work perfectly. Hope this helps.

**Comment by Simonwep** [comment]
~~Huh, now this is strange, the value of zIndex should be an integer as mentioned [here](https://www.w3.org/TR/1998/REC-CSS2-19980512/visuren.html#propdef-z-index) in the official w3 draft 🤔~~

~~Is there any way to get hand on a reproduction in a JSFiddle? Because, even if the type for zIndex is still integer, if you use `style.zIndex = '1'` gives the same result as `style.zIndex = 1`.~~

Okay this is an actual a bug in the `.css` function. If I pass a integer in it it will get [unified](https://github.com/Simonwep/selection/blob/master/src/utils.js#L58) (with `px` as standard). So it'd result in `zIndex: 1px` which is invalid.

I'll fix this later today and roll out `1.4.1`, I should pay attention to how my own utils work 😅

**Comment by Simonwep** [comment]
@dsechin Fixed in 5539f8b :) I've published [`1.4.1`](https://github.com/Simonwep/selection/releases/tag/1.4.1) where this fix is applied!

**Comment by dsechin** [comment]
> @dsechin Fixed in [5539f8b](https://github.com/Simonwep/selection/commit/5539f8bf2089a5f076a5809768976c4f24ae422e) :) I've published [`1.4.1`](https://github.com/Simonwep/selection/releases/tag/1.4.1) where this fix is applied!

Thank you for the quick fix! ^_^
